frameworks/kafka/docs/managing.md

```diff
@@ -140,6 +140,59 @@ Increase the `BROKER_COUNT` value via the DC/OS web interface as in any other co
 
         $ dcos package install kafka -—options=options.json
 
+# Graceful Shutdown
+## Extend Kill Grace Period
+
+Increase the `brokers.kill_grace_period` value via the DC/OS CLI, i.e. to `60`
+seconds. This example assumes that the Kafka service instance is named `kafka`.
+
+During the reconfiguration, each of the Kafka broker tasks are restarted. During
+the shutdown portion of the task restart, the previous configuration value for
+`brokers.kill_grace_period` is in effect. Following the shutdown, each broker
+task is launched with the new effective configuration value. So, care should be
+taken to monitor the period of time that Kafka brokers are taking to cleanly
+shutdown. The relevant log entries are enumerated within [Configure](configure.md).
+
+Create an options file `kafka-options.json` with the following content:
+
+        {
+            "brokers": {
+                "kill_grace_period": 60
+            }
+        }
+
+Issue the following command:
+
+        dcos kafka --name=/kafka update --options=kafka-options.json
+
+## Restart a Broker with Grace
+
+When restarting a Kafka broker, when the Kill Grace Period has been configured
+to a period that sufficiently allows Kafka to cleanly shutdown, the broker's
+shutdown time will be longer, but the subsequent startup time will be much more
+rapid as the Byzantine reconciliation activities otherwise required for startup
+can be safely skipped. Besides the observable differences in shutdown and startup
+times, broker task restart should otherwise be the same as when the broker is not
+granted sufficient grace.
+
```

Comment:
An appropriate kill grace period has been configured allows Kafka to shut down cleanly. A graceful (or clean) shutdown takes longer than an ungraceful shutdown, but the next startup will be much quicker. This is because the complex reconciliation activities that would have been required are not necessary after graceful shutdown.